examples/keycloak_oauth.rs

```diff
@@ -0,0 +1,303 @@
+/// This example shows how to use pgwire with Keycloak OAuth.
+/// To connect with psql:
+/// 1. Install libq-oauth: sudo apt-get install libpq-oauth
+/// 2. Setup keycloak. check this for more details: https://habr.com/en/companies/tantor/articles/959776/
+/// 3. Execute: psql "postgres://postgres@localhost:5432/postgres?oauth_issuer=http://localhost:8080/realms/postgres-realm&oauth_client_id=postgres-client&oauth_client_secret=<my-client-secret>"
+use base64::prelude::BASE64_URL_SAFE_NO_PAD;
+use base64::Engine;
+use jsonwebtoken::{decode, decode_header, Algorithm, DecodingKey, Validation};
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::{BufReader, Error as IOError, ErrorKind};
+use std::sync::Arc;
+
+use async_trait::async_trait;
+
```

Comment:
we can reorganize the imports, first section for std, second section for other deps